src/include/processor/operator/recursive_extend/path_scanner.h

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include <stack>
+
+#include "bfs_state.h"
+
+namespace kuzu {
+namespace processor {
+
+/*
+ * PathScanner Design
```

Comment:
Try to work harder on these comments. You are not giving enough details and context on some of your comments, e.g., "We scan from dst to src because semi mask might be available and we only need to scan a subset of all dst nodes." You are expecting the reader to infer a lot much from little detail.

I'd remove it all. First you can put this comment on top of PathScanner (not FixedLengthPathScanner).
```
 * Variable-length joins return union of paths with different length (e.g. *2..3). Note that we only keep 
 * track of the backward edges between the frontier in the RecursiveJoin operator (these frontiers are stored in the BaseBFSMorsel that was used to keep the data related to the BFS that was computed in the RecursiveJoin). Therefore we cannot start 
 * from the src and traverse to find all paths of all lengths. We can only start from nodes in a particular 
 * frontier and traverse backwards to the source. But whenever we start from a particular frontier, say 
 * the k'th frontier, we can only traverse paths of length k. Therefore, PathScanner scans 
 * these paths length by length, i.e. we first scan all length-2 paths, then scan all length-3 paths. 
```

Then here, you can put this comment:

```
 * FixedLengPathScanner scans all paths of a fixed length k. This is done by starting a backward
 * traversals from only the destination nodes in the k'th frontier (assuming the first frontier has
 * index 0) over the backwards edges stored between the frontiers stored in the BFSMorsel that
 * was used to store the data related to the BFS that was computed in the RecursiveJoin operator.
 * To identify the destination nodes in the k'th frontier, we use a semi mask that marks the destination 
 * nodes (or the BFSMorsel::isAllDstTarget() function which indicates that every node is a possible
 * destination).
```